File: app/src/main/java/temple/edu/yelp_randomizer/restaraunts/CategoriesAdapter.java
package temple.edu.yelp_randomizer.restaraunts;

import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class CategoriesAdapter extends BaseAdapter {

    Context context;
    ArrayList<String> categories;

    public CategoriesAdapter(Context context, ArrayList<String> categories){
        this.context = context;
        this.categories = categories;
    }
    @Override
    public int getCount() {
        return categories.size();
    }

    @Override
    public Object getItem(int i) {
        return categories.get(i);
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        TextView textView;

        if (view != null){
            textView = (TextView) view;
        }
        else{
            textView = new TextView(context);
        }
        textView.setLines(4);
        textView.setGravity(Gravity.CENTER);
        textView.setText(categories.get(i).toUpperCase());

        return textView;
    }
}

File: app/src/main/java/temple/edu/random/restaraunts/CategoriesAdapter.kt
package temple.edu.random.restaraunts

import android.content.Context
import android.widget.BaseAdapter
import android.view.ViewGroup
import android.widget.TextView
import android.view.Gravity
import android.view.View
import java.util.ArrayList

class CategoriesAdapter(var context: Context, var categories: ArrayList<String>) : BaseAdapter() {
    override fun getCount(): Int {
        return categories.size
    }

    override fun getItem(i: Int): Any {
        return categories[i]
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        val textView: TextView
        textView = if (view != null) {
            view as TextView
        } else {
            TextView(context)
        }
        textView.setLines(4)
        textView.gravity = Gravity.CENTER
        textView.text = categories[i].toUpperCase()
        return textView
    }
}